index.html

```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+    <style>
+        html, body {height: 100%; margin: 5px; padding: 0;}
+        table {background-color: linen; font-family: "Verdana", Times, serif;}
+        th {background-color: #4CAF50;color: white;}
+        tr:nth-child(even) {background-color: #f2f2f2}
+        p {font-family: "Verdana", Times, serif;}
+    </style>
+    <link rel="stylesheet" type="text/css" href="mapStyle.css">
+	<meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>ArcTouch Code Challenge</title>
+    <script type="text/javascript" src="prototype.js"></script>
+    <script type="text/javascript" src="utils.js"></script>
+    <script type="text/javascript" src="map.js"></script>
+    <script type="text/javascript">
+        //<![CDATA[
+
+        function getPreviousRouteList() {
+            var street = getURLParameterByName("street");
+            if (street != null) {
+                document.getElementById("txtStreetName").value = street;
+                getRouteListByStreetName();
+            }
+        }
+
+        function getRouteListByStreetName() {
+            var txtStreetName = document.getElementById("txtStreetName").value.replace(/^\s+|\s+$/gm,'');

```

Comment:
This function has a lot of mixed responsibilities. It reads the input from the UI, validates it and does a network request. How would you do to fix that?